src/devices/cpu/z80/z80.cpp

```diff
@@ -476,6 +481,51 @@ void z80_device::illegal_2()
 			m_opcodes.read_byte((PC - 1) & 0xffff));
 }
 
+u8 z80_device::flags_szyxc(u16 value)
+{
+	u8 f = flags_sz(value);
+	f |= (value >> 8) & CF;
+	return f;
+}
+
+u8 z80_device::flags_sz(u8 value)
+{
+	u8 f = value & (SF | YF | XF);  // SF + undocumented flag bits 5+3
+	f |= value ? 0 : ZF;
+	return f;
+}
+
+u8 z80_device::flags_sz_bit(u8 value)
+{
+	u8 f = value & (SF | YF | XF);  // SF + undocumented flag bits 5+3
+	f |= value ? 0 : (ZF | PF);
+	return f;
+}
+
+u8 z80_device::flags_szp(u8 value)
+{
+	u8 f = flags_sz(value);
+	f |= __builtin_parity(value) ? 0 : PF;
```

Comment:
Any concerns about this intrinsic?

Reply:
Yes, it needs to be portable.
We have population_count_32(val) & 1, and it'll be quite slow if the user's CPU doesn't support a built-in method.

Reply:
Sure, I'll replace it with "math"

Reply:
done